Bottom: 05e8f43e60ef29b0be5ecb4cc8bc13ea5a348fa3
Top:    cd08fed68bccc971c2a6d56b1ed4cd8be4b92f15
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-10-01 12:51:59 +0100

action.c: Fix the SetValue(TextScale, ...) action after nanometer conversion


---

diff --git a/src/action.c b/src/action.c
index 62c53e3..2a0be9e 100644
--- a/src/action.c
+++ b/src/action.c
@@ -2173,6 +2173,7 @@ ActionSetValue (int argc, char **argv, Coord x, Coord y)
   char *units = ARG (2);
   bool absolute;			/* flag for 'absolute' value */
   double value;
+  int text_scale;
   int err = 0;
 
   if (function && val)
@@ -2217,8 +2218,10 @@ ActionSetValue (int argc, char **argv, Coord x, Coord y)
 
 	case F_Text:
 	case F_TextScale:
-	  value /= 45;
-	  SetTextScale (absolute ? value : value + Settings.TextScale);
+	  text_scale = value / (double)FONT_CAPHEIGHT * 100.;
+	  if (!absolute)
+	    text_scale += Settings.TextScale;
+	  SetTextScale (text_scale);
 	  break;
 	default:
 	  err = 1;
